### PR TITLE
fix: allow SQL objects names with unicode char and is not limited to …

### PR DESCRIPTION
**Pull request by amanPras**
## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Check added usecase test
```cobol
       IDENTIFICATION DIVISION.
       PROGRAM-ID. test232.
       ENVIRONMENT DIVISION.
       CONFIGURATION SECTION.
       SPECIAL-NAMES.
       INPUT-OUTPUT SECTION.
       FILE-CONTROL.
       DATA DIVISION.
        FILE SECTION.
        WORKING-STORAGE SECTION.
        01 AB-FR pic x.
        01 A-MD pic x.
        LINKAGE SECTION.
       PROCEDURE DIVISION.
000000     EXEC SQL
000000         SELECT FÖLJENR
000000         INTO :A-FR
000000         FROM D51.TFÖLJESR
000000         WHERE FÖLJENR = :A-FR
000000           AND MTRLID = :A-MD
000000         WITH UR
000000     END-EXEC
``` 

## Checklist:
- [x] Each of my commits contains one meaningful change
- [x] I have performed rebase of my branch on top of the development
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: server/engine/src/main/antlr4/org/eclipse/lsp/cobol/implicitDialects/sql/Db2SqlLexer.g4 [hunk 1]
```diff
@@ -857,7 +857,7 @@ DATELITERAL: '\'' (DIGIT DIGIT DIGIT DIGIT '-' DIGIT DIGIT '-' DIGIT DIGIT | //y
 
 SINGLEDIGITLITERAL : DIGIT;
 INTEGERLITERAL : DIGIT+;
-IDENTIFIER : [a-zA-Z0-9][_a-zA-Z0-9-]*;
+IDENTIFIER : [\p{Alpha}\p{General_Category=Other_Letter}] [\p{Alnum}\p{General_Category=Other_Letter}]*;
```
Comment (ap891843) [inline server/engine/src/main/antlr4/org/eclipse/lsp/cobol/implicitDialects/sql/Db2SqlLexer.g4:860]: This rule consists of two main parts:
\p{Alpha}: Matches any alphabetic character (letters from any script)
\p{General_Category=Other_Letter}: Matches characters categorized as "Other_Letter" in Unicode

References:
https://github.com/antlr/antlr4/blob/master/doc/unicode.md
https://github.com/antlr/antlr4/blob/master/doc/lexer-rules.md#lexer-rule-elements

File: server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/sql/TestSqlObjectsName.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+package org.eclipse.lsp.cobol.usecases.sql;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.eclipse.lsp.cobol.test.engine.UseCaseEngine;
+import org.junit.jupiter.api.Test;
+
+/** Test SQL objects names with unicode char not in [A-z0-9] doesn't break parsing */
+public class TestSqlObjectsName {
+  public static final String TEXT =
```
Comment (amanPras) [inline server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/sql/TestSqlObjectsName.java:24]: I am sure this is not a compliable code . But added it to allow diff chars

File: server/engine/src/main/antlr4/org/eclipse/lsp/cobol/implicitDialects/sql/Db2SqlLexer.g4 [hunk 3]
```diff
@@ -857,7 +857,7 @@ DATELITERAL: '\'' (DIGIT DIGIT DIGIT DIGIT '-' DIGIT DIGIT '-' DIGIT DIGIT | //y
 
 SINGLEDIGITLITERAL : DIGIT;
 INTEGERLITERAL : DIGIT+;
-IDENTIFIER : [a-zA-Z0-9][_a-zA-Z0-9-]*;
+IDENTIFIER : [\p{Number}\p{Alpha}\p{General_Category=Other_Letter}] [-_\p{Number}\p{Alnum}\p{General_Category=Other_Letter}]*;
```
Comment (amanPras) [inline server/engine/src/main/antlr4/org/eclipse/lsp/cobol/implicitDialects/sql/Db2SqlLexer.g4]: This rule consists of two main parts:
\p{Alpha}: Matches any alphabetic character (letters from any script)
\p{General_Category=Other_Letter}: Matches characters categorized as "Other_Letter" in Unicode

References:
https://github.com/antlr/antlr4/blob/master/doc/unicode.md
https://github.com/antlr/antlr4/blob/master/doc/lexer-rules.md#lexer-rule-elements
https://unicode.org/reports/tr44/#General_Category_Values

Comment (slavek-kucera) [reply server/engine/src/main/antlr4/org/eclipse/lsp/cobol/implicitDialects/sql/Db2SqlLexer.g4]: is `\p{Number}\p{Alpha}` different from `\p{Alnum}`?
is `\p{Number}\p{Alnum}` the same as `\p{Alnum}`?

Comment (ap891843) [reply server/engine/src/main/antlr4/org/eclipse/lsp/cobol/implicitDialects/sql/Db2SqlLexer.g4:860]: Yes, it was weird , it didn't match just number. For e.g. `100` was not matched to Alpha

**Review by slavek-kucera — APPROVED** [review]

**Pull request merged**